020-Getting-started/050-sql-access.mdx

```diff
@@ -0,0 +1,167 @@
+| file | jsonb | validated by Xata |
+| filearray | jsonb | validated by Xata |
+| link | foreign key | |
+
+Database tables are stored in PostgreSQL. Linked columns are foreign
+keys pointing to the linked table. If you want to access a linked
+column, use JOINs or WHERE conditions to find the corresponding
+records.  Example: `SELECT "Users".email, address.city FROM "Users"
+LEFT JOIN address ON "Users".id=address.id `.
+
+Other special Xata datatypes like `email` require you to insert it
+using the Xata API or SDK, otherwise Xata cannot check if the data
+inserted is valid or not. Thus, we suggest you do not write such
+column directly. Instead use the API we provide.
+
+## Endpoint
+
+`POST https://{your-workspace-slug}.{region}.xata.sh/db/db_branch_name/sql`
+
+Available parameters:
+
+* `query`: required, your SQL statement.
+* `params`: list of parameters when submitting a parameterized query.
+*  `consistency`:  consistency  when   executing  a  query.  Available options: `strong`, `eventual`. Default: `strong`.
+
+## Examples
+
+### Fetch all records
+
+````ts|python|json
+  ```ts
+  const { records: myRecord } = await xata.sql<TeamsRecord>`SELECT * FROM teams`;
+  ```
+  ```python
+  records = xata.sql().query("SELECT * FROM \"teams\"")
+  ```
+  ```json
+  {
+	  "query": "SELECT * FROM \"teams\""
+  }
+  ```
+````
+
+The following response is returned:
+
+```json
+{
+  "records": [
+    {
+      "id": "rec_c8hng2h26un90p8sr7k0",
+      "name": "Matrix",
+      "owner": {
+        "id": "myid"
+      },
+      "xata": {
+        "version": 0,
+        "createdAt": "2023-05-15T08:21:31.96526+01:00",
+        "updatedAt": "2023-05-15T21:58:54.072595+01:00"
+      }
+    }
+  ]
+}
+```
+
+### Select a subset of columns from your table
+
+````ts|python|json
+  ```ts
+  const { records: myRecord } = await xata.sql<TeamsRecord>`SELECT name, city FROM "Users"`;
+  ```
+  ```python
+  records = xata.sql().query("SELECT name, city FROM \"Users\"")
+  ```
+  ```json
+  {
+	  "query": "SELECT name, city FROM \"Users\""
+  }
+  ```
+````
+
+Response:
+
+```json
+{
+  "records": [
+    {
+      "city": "New York",
+    	"name": "Keanu Reaves"
+    }
+  ]
+}
+```
+
+### Insert a new record using a parameterized query
+
+````ts|python|json
+  ```ts
+  const { records: myRecord } = await xata.sql<TeamsRecord>`INSERT INTO "Users" (name, email) VALUES ($1, $2)`;
+  ```
+  ```python
+  records = xata.sql().query(
+    "INSERT INTO \"Users\" (name, email) VALUES ($1, $2)",
+    ["Keanu Reeves", "keanu@example.com"]
+  )
+  ```
+  ```json
+  {
+	  "query": "INSERT INTO \"Users\" (name, email) VALUES ($1, $2)",
+	  "params": ["Keanu Reeves", "keanu@example.com"]
+  }
+  ```
+````
+
+## Limitations
+
+* `xata.*` internal fields are read-only.
+
+* You can only submit one statement per request.
+
+* Xata returns 1000 records at a time. Please access your data using
+  `LIMIT` and `OFFSET` keywords to iterate over your data.
```

Comment:
Can we clarify if/how using OFFSET allows to access more than the first 1000 records?